force-app/main/default/classes/ODRIntegrationTest.cls

```diff
@@ -335,4 +335,71 @@ public class ODRIntegrationTest {
 
         System.assertEquals(new Map<String, Integration_Log__c>(), m1);
     }
+    
+    @isTest
+    public Static void fetchPrescriptionHistoryWithSearchKey_Test(){
+        Drug__c drug = TestFactory.newDrug('Drug 1');
+        insert drug;
+        
+        Account patient = TestFactory.newPatient('Patient 1');
+        Account provider = TestFactory.newProvider('Provider 1');
+        insert new Account[] { patient, provider };
+            
+        Case saCase = TestFactory.newCase(drug, patient, provider, 'Test case 1');
+        insert saCase;
+        
+        Test.setMock(HttpCalloutMock.class, new RestMock());
+        Test.startTest();
+        PrescriptionHistoryResponse resObjWithSearchKey = ODRIntegration.fetchPrescriptionHistoryWithSearchKey(saCase.id, '1', '1', new List<Integer>(), 'CONTRACEPTION', '10');
+        PrescriptionHistoryResponse resObjWithOutSearchKey = ODRIntegration.fetchPrescriptionHistoryWithSearchKey(saCase.id, '1', '1', new List<Integer>(), 'Test', '10');
+        Test.stopTest();
+        Assert.areEqual(resObjWithSearchKey.medHistory.medRecords.size(), 1, 'Search Failed');
+        Assert.areEqual(resObjWithOutSearchKey.medHistory.medRecords.size(), 0, 'Search Failed for searchKey with is not in response generic name');
+    }
+    
+    @isTest
+    public Static void fetchPrescriptionHistoryWithSearchKey_Test_Negative(){
+        Drug__c drug = TestFactory.newDrug('Drug 1');
+        insert drug;
+        
+        Account patient = TestFactory.newPatient('Patient 1');
+        Account provider = TestFactory.newProvider('Provider 1');
+        insert new Account[] { patient, provider };
+            
+        Case saCase = TestFactory.newCase(drug, patient, provider, 'Test case 1');
+        insert saCase;
+        
+        Test.setMock(HttpCalloutMock.class, new RestMock_Negative());
+        Test.startTest();
+        PrescriptionHistoryResponse resObjWithSearchKey = ODRIntegration.fetchPrescriptionHistoryWithSearchKey(saCase.id, '1', '1', new List<Integer>(), 'CONTRACEPTION', '10');
+        Test.stopTest();
+        Assert.areNotEqual(resObjWithSearchKey, NULL, 'Response is blank');
```

Comment:
Thanks for making the change. Instead of increasing the lines of code, i was hoping you would add the negative scenario in the test method above `fetchPrescriptionHistoryWithSearchKey_Test`. Tech-debt item.